src/components/PopupWithForm.js

```diff
@@ -22,9 +23,31 @@ export default class PopupWithForm extends Popup {
     this._modalForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
       const inputValues = this._getInputValues();
-      console.log("Input values before submit:", inputValues);
+      this.renderSaving(true);
       this._handleFormSubmit(inputValues);
+      this.renderSaving(false);
     });
     super.setEventListeners();
   }
+
+  stopSaving(popup) {
+    popup._modalForm.querySelector(".modal__save-button").textContent =
+      popup._buttonText;
+    console.log("Found _modalForm and done");
+    popup.close();
+  }
+
+  renderSaving(isSaving) {
+    if (isSaving) {
+      this._modalForm.querySelector(".modal__save-button").textContent =
+        "Saving...";
+      console.log("Found _modalForm and saving");
+      console.log(
+        this._modalForm.querySelector(".modal__save-button").textContent
+      );
+    } else {
+      console.log(this);
+      setTimeout(this.stopSaving, 1000, this);
```

Comment:
`setTimeout` is not needed because you should have **natural** waiting for the server response

Reply:
I fixed this, but it now closes so quickly that I don't see the change in button text.

Reply:
> I fixed this, but it now closes so quickly that I don't see the change in button text.


* In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 



In `index.js` you need to fix it everywhere